core/domain/user_domain.py

```diff
@@ -107,3 +111,133 @@ def to_dict(self):
+
+    def add_exploration_id(self, exploration_id):
+        """Adds the exploration id to the list of incomplete exploration ids."""
+
+        self.incomplete_exploration_ids.append(exploration_id)
+
+    def remove_exploration_id(self, exploration_id):
+        """Removes the exploration id from the list of incomplete exploration
+        ids.
+        """
+
+        self.incomplete_exploration_ids.remove(exploration_id)
+
+    def get_last_playthrough_information(self, exploration_id):
+        """Get the information regarding the last playthrough of the user of
+        the given exploration.
+
+        Args:
+            exploration_id: str. The id of the exploration for which we have to
+            get the last playthrough information.
+
+        Returns:
+            ExpUserLastPlaythrough. A ExpUserLastPlaythrough domain object
+            corresponding to the given exploration id.
+        """
+        incomplete_exploration_user_model = (
+            user_models.ExpUserLastPlaythroughModel.get(
+                self.id, exploration_id))
+
+        if not incomplete_exploration_user_model:
+            incomplete_exploration_user_model = (
+                user_models.ExpUserLastPlaythroughModel.create(
+                    self.id, exploration_id))
+
+        return ExpUserLastPlaythrough(
+            self.id, exploration_id,
+            incomplete_exploration_user_model.version_last_played,
+            incomplete_exploration_user_model.time_last_played_msec,
+            incomplete_exploration_user_model.last_state_played)
+
+    def delete_last_playthrough_information(self, exploration_id):
+        """Deletes the last playthrough information model corresponding to the
+        given user id and exploration id.
+
+        Args:
+            exploration_id: str. The id of the exploration for which we have to
+            delete the last playthrough information model.
+        """
+        incomplete_exploration_user_model = (
+            user_models.ExpUserLastPlaythroughModel.get(
+                self.id, exploration_id))
+
+        incomplete_exploration_user_model.delete()
+
+
+class ActivitiesCompletedByLearner(object):
+    """Domain object for the activities completed by learner model."""
+
+    def __init__(self, user_id, completed_exploration_ids,
+                 completed_collection_ids):
+        self.id = user_id
+        self.completed_exploration_ids = completed_exploration_ids
+        self.completed_collection_ids = completed_collection_ids
+
+    def add_exploration_id(self, exploration_id):
+        """Adds the exploration id to the list of completed exploration ids."""
+
+        self.completed_exploration_ids.append(exploration_id)
+
+    def add_collection_id(self, collection_id):
+        """Adds the collection id to the list of completed collection ids."""
+
+        self.completed_collection_ids.append(collection_id)
+
+
+class IncompleteCollections(object):
```

Comment:
Suggest putting this adjacent to IncompleteExplorations, rather than separating them by an unrelated class.